### PR TITLE
codeintel-qa: Rewrite precise-code-intel-tester

### LINKED ISSUE #25387: Flake: code-intel-qa disabled
- **Name of test:** code-intel-qa
- **Example failure:**
https://buildkite.com/sourcegraph/code-intel-qa/builds/4847#296f3dc7-34cf-46f4-acb9-3b1f24a2fce7
https://buildkite.com/sourcegraph/code-intel-qa/builds/4845#ecff50fc-a139-482a-89ca-e6fb8839d0dd

Screenshot 2021-09-27 at 12 17 12

- **PR**: https://github.com/sourcegraph/sourcegraph/pull/25386/files

#### Additional details

https://sourcegraph.slack.com/archives/CMBA8F926/p1632733591338100

### PR DESCRIPTION
**Pull request by efritz**
Fixes #25387. This makes several improvements to the codeintel-qa pipeline, including:

- simplified concurrency management in the test code
- simplified infrastructure - no longer needs baremetal queue or vagrant to run
- fixes to the underlying instance to reduce flake (see also #25571 and #25578)

### CODE REVIEW

**Comment by Numbers88s** [comment]
LGTM

**Review by eseliger — APPROVED** [review]
Rubber stamping, as we talked through this quite a bit already and overall lgtm.

File: dev/codeintel-qa/cmd/query/main.go [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+		select {
+		case err, ok := <-errCh:
+			if ok {
+				return err
+			}
+
+			break loop
+
+		case <-time.After(time.Second):
+			if verbose {
```
Comment (jhchabran) [inline dev/codeintel-qa/cmd/query/main.go:89]: Shouldn't this be `!verbose`? As it's now, it'll skip intermediate results if the verbose flag is set. 

```suggestion
			if !verbose {
```

**Review by jhchabran — APPROVED** [review]
LGTM (you probably want to check the minor comment about the verbose flag)

File: dev/codeintel-qa/cmd/query/main.go (hunk 1 above)
Comment (efritz) [reply dev/codeintel-qa/cmd/query/main.go:89]: We print verbose output differently closer to the GraphQL call, so we disable it here.

**Pull request merged**